workspace/src/path_planning/path_planning/path_planning.py

```diff
@@ -59,12 +59,21 @@ def __init__(self):
         package_share_directory = get_package_share_directory('path_planning')
 
         # READ IN PARAMETERS
+        self.declare_parameter('planning_mode', "cones")
+        self.planning_mode = self.get_parameter('planning_mode').get_parameter_value().string_value
+
         self.declare_parameter('vis', False)
         self.vis = self.get_parameter('vis').get_parameter_value().bool_value
 
         self.declare_parameter('lookahead', 2.0)
         self.lookahead = self.get_parameter('lookahead').get_parameter_value().double_value
         
+        if(self.planning_mode == "waypoints"):
+            self.declare_parameter('path_file', "/home/art/art/workspace/src/path_planning/path_planning/path.csv")
+            path_file = self.get_parameter('path_file').get_parameter_value().string_value
+            file = open(path_file)
```

Comment:
you should use a context manager when opening files. I don't think file is ever closed here, right? or does np.loadtxt do this under the hood? eitherway, unclear if that's the case. I think loadtxt can just take the string pathname. 

this will probs change if you instead use parameters tho